projects/RabbitMQ.Client.OAuth2/OAuth2Client.cs

```diff
@@ -169,127 +114,162 @@ internal class OAuth2Client : IOAuth2Client, IDisposable
         private readonly string _clientId;
         private readonly string _clientSecret;
         private readonly Uri _tokenEndpoint;
-        private readonly string _scope;
+        private readonly string? _scope;
         private readonly IDictionary<string, string> _additionalRequestParameters;
 
         public static readonly IDictionary<string, string> EMPTY = new Dictionary<string, string>();
 
         private HttpClient _httpClient;
 
-        public OAuth2Client(string clientId, string clientSecret, Uri tokenEndpoint, string scope,
-                IDictionary<string, string> additionalRequestParameters,
-                HttpClientHandler httpClientHandler)
+        public OAuth2Client(string clientId, string clientSecret, Uri tokenEndpoint,
+            string? scope,
+            IDictionary<string, string>? additionalRequestParameters,
+            HttpClientHandler? httpClientHandler)
         {
-            this._clientId = clientId;
-            this._clientSecret = clientSecret;
-            this._scope = scope;
-            this._additionalRequestParameters = additionalRequestParameters == null ? EMPTY : additionalRequestParameters;
-            this._tokenEndpoint = tokenEndpoint;
+            _clientId = clientId;
+            _clientSecret = clientSecret;
+            _scope = scope;
+            _additionalRequestParameters = additionalRequestParameters == null ? EMPTY : additionalRequestParameters;
+            _tokenEndpoint = tokenEndpoint;
 
             _httpClient = httpClientHandler == null ? new HttpClient() :
                 new HttpClient(httpClientHandler);
```

Comment:
Thanks for pointing this out. I'm going to change this code.

Reply:
Given this targets a new version and this is a dedicated package would it make sense to depend on Microsoft.Extensions.Http and allow setting an HttpClientFactory? But maybe this is too much for now since the package relies on hosting abstractions quite heavily

Reply:
The method is now removed. Is that a good idea? How would anyone that has some proxy requirements etc configure now the underlying HTTP stuff?

Reply:
> The method is now removed. Is that a good idea? How would anyone that has some proxy requirements etc configure now the underlying HTTP stuff?

Thanks for explaining one reason why the code existed the way it did 😄 I'll revert that change.

Reply:
> But maybe this is too much for now since the package relies on hosting abstractions quite heavily

I don't know what this statement means ... 🤔

Reply:
He's referring to IHttpClientFactory, which unfortunately lives in one of those Nuget Microsoft.Hosting.* packages. The Nuget package tail is quite long for these. Including mandatory DI usage afaik.

Reply:
Ah, thanks! I see that here, yikes - https://www.nuget.org/packages/microsoft.extensions.http#dependencies-body-tab

Reply:
The usual way of working around this issue is providing an abstraction for that.

RabbitMQ can have its own `IHttpClientFactory` and users (or another NuGet package) can provide an implementation based on `System.Net.Http.IHttpClientFactory`.

That's how `Newtonsoft.Json` allows to use array pools: https://www.newtonsoft.com/json/help/search.html?SearchText=arraypool

Reply:
We'll see if anyone asks for a feature like that going forward. Thanks for everyone's input.